clients/pkg/promtail/targets/kafka/target.go

```diff
@@ -90,24 +92,18 @@ func (t *Target) run() {
 			out = out.Merge(lbs)
 		}
 
-		logLines := []string{string(message.Value)}
-
-		if t.messageParser != nil {
-			ll, err := t.messageParser(message)
-			if err != nil {
-				level.Error(t.logger).Log("msg", "message parsing error", "err", err)
-			} else {
-				logLines = ll
-			}
-		}
-
-		for _, logLine := range logLines {
-			t.client.Chan() <- api.Entry{
-				Entry: logproto.Entry{
-					Line:      logLine,
-					Timestamp: timestamp(t.useIncomingTimestamp, message.Timestamp),
-				},
-				Labels: out,
+		logLines, err := t.messageParser.Parse(message)
+		if err != nil {
+			level.Error(t.logger).Log("msg", "message parsing error", "err", err)
```

Comment:
nit: can we do a `continue` after the log in the if branch, and do the `for _, logLine := range logLines { ...` outside the else statement? Might look more idiomatic

Reply:
You are right. I was thinking of doing it as well.

I did it differently because I think, in this case, we will have to duplicate also `t.session.MarkMessage(message, "")` statement before `continue`. Because if we can not process an event, we want to move to the next one.

This is why I decided to go with `else`.

But I don't have any strong opinion here 😃 

Would you prefer to go with `continue` and a `t.session.MarkMessage(message, "")`?

Reply:
Ohh you are right, I forgot about the message ACK. Then it's good like this!